spec/views/posts/post_index_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe 'posts#index', type: :feature do
+  before(:each) do
+    @first_user = User.create(name: 'Kim', photo: 'https://picsum.photos/200/300', bio: 'I am a Software Engineer')
+    @first_post = Post.create(author_id: @first_user.id, title: 'My first post', text: 'This is my first post')
+    @second_post = Post.create(author_id: @first_user.id, title: 'My second post', text: 'This is my second post')
+    # secod user
+    @second_user = User.create(name: 'Sarah', photo: 'https://picsum.photos/200/300', bio: 'I am a teacher')
+    @second_user_first_post = Post.create(author_id: @second_user.id, title: 'My first post',
+                                          text: 'This is my first post')
+
+    Comment.create(author_id: @second_user.id, post_id: @second_user_first_post.id, text: 'This is my first comment')
+
+    visit user_posts_path(@user.id)
+  end
+
+  it 'can see the user profile picture.' do
+    expect(page).to have_css("img[src*='https://picsum.photos/200/300']")
+  end
+
+  it 'I can see the user username' do
+    expect(page).to have_content('Kim')
+  end
+
+  it 'can see the number of posts the user has written' do
+    expect(page).to have_content('2')
+  end
+
+  it 'I can see a post title' do
+    expect(page).to have_content('This is my first post')
+    expect(page).to have_content('This is my second post')
+  end
+
+  it 'can see some of the post body' do
+    expect(page).to have_content('This is my first post')
+    expect(page).to have_content('This is my second post')
+  end
+
+  it 'can see how many comments a post has' do
+    expect(page).to have_content('1')
+  end
+
+  it 'can see how many likes a post has.' do
+    expect(page).to have_content('0')
+  end
+  it 'can see a button to create a new post' do
+    expect(page).to have_link('Create new post')
+  end
+
+  it 'When I click on a post, it redirects me to that post show page.' do
+    click_link 'This is my second post'
```

Comment:
- [ ] Great job working on the tests so far but after running your project I think your tests are not passed so please make sure to have all your tests passed I face this problem he show that undefined method `id' for nil:NilClass 
visit user_posts_path(@user.id) since you considered the id as a method🙏🏻💯 so please check the problem and I think it is the same problem in all your tests so please make sure to solve it 🙏🏻🙏🏻

## The error happed ❗❗
![image](https://user-images.githubusercontent.com/58553711/227404284-e959cd16-9123-4b91-bfee-8fc8269cd19e.png)

### In this files, they all have the same problem couldn't identify for the id in `user.id` !?

![image](https://user-images.githubusercontent.com/58553711/227408048-c9ef2638-485d-4e34-8df3-cd5c4a1a235d.png)